PictureApp.API/Controllers/FollowersController.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Security.Authentication;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using PictureApp.API.Extensions.Exceptions;
+using PictureApp.API.Services;
+
+namespace PictureApp.API.Controllers
+{
+    [Authorize]
+    [Route("api/users/{userId}/followers")]
+    [ApiController]
+    public class FollowersController : ControllerBase
+    {
+        private readonly IUserService _userService;
+        private readonly IFollowerService _followerService;
+
+        public FollowersController(IUserService userService, IFollowerService followerService)
+        {
+            _userService = userService;
+            _followerService = followerService;
+        }
+
+        [HttpPost("{id}/setfollow")]
+        public async Task<IActionResult> SetUpFollower(int userId, int id)
+        {
+            if (userId != int.Parse(User.FindFirst(ClaimTypes.NameIdentifier).Value))
+                return Unauthorized();
+
+            try
+            {
+                await Task.Run(() => _followerService.SetUpFollower(userId, id));
+
+                return Ok();
+            }
+            catch (EntityNotFoundException)
+            {
+                return BadRequest("Can't set up follower");
+            }
+        }
+
+        [HttpPost("{id}/setunfollow")]
+        public async Task<IActionResult> SetUpUnFollow(int userId, int id)
+        {
+            if (userId != int.Parse(User.FindFirst(ClaimTypes.NameIdentifier).Value))
```

Comment:
I would like to put this authorization checking to the separate private method. The same with similar code in action above.

Reply:
good point, fixed by:
https://github.com/anowakowski/PictureApp_API/pull/33/commits/d75fbfb18e6a7c0b48ad945c5f31adcc42df0776